test/wanda_web/controllers/executions_controller_test.exs

```diff
@@ -0,0 +1,28 @@
+defmodule WandaWeb.ExecutionsControllerTest do
+  use WandaWeb.ConnCase, async: true
+  import OpenApiSpex.TestAssertions
+  import Wanda.Factory
+
+  alias Wanda.Results.ExecutionResult
+  alias WandaWeb.ApiSpec
+
+  describe "Executions controller" do
+    test "listing check executions produces a CheckExecutionResponse", %{conn: conn} do
+      [
+        %ExecutionResult{},
+        %ExecutionResult{},
+        %ExecutionResult{},
+        %ExecutionResult{},
+        %ExecutionResult{}
+      ] = insert_list(5, :execution_result)
+
+      json =
+        conn
+        |> get("/api/checks/executions")
+        |> json_response(200)
+
+      api_spec = ApiSpec.spec()
+      assert_schema(json, "CheckExecutionResponse", api_spec)
```

Comment:
I think you could assert the returned json value with an expected one

Reply:
Maybe some punctual assertions, the expected one I think it would be very big and with little value because the meaningful test is the one we do on the context. Here's we're testing more that the response is compliant with the spec. But valid point, I'll think about it